collated/main/CindyTsai1.md

```diff
@@ -0,0 +1,536 @@
+
+        for (int i = 0; i < predicate.size() - 1; i++) {
+            String predicates = predicate.get(i);
+
+            if (predicates.equals(PREFIX_NAME.getPrefix())) {
+                ArrayList<String> nameList = findName(predicate.get(i + 1));
+                for (int j = 0; j < nameList.size(); j++) {
+                    if (!predicateList.contains(nameList.get(j))) {
+                        predicateList.add(nameList.get(j));
+                    }
+                }
+            }
+
+            if (predicates.equals(PREFIX_PHONE.getPrefix())) {
+                ArrayList<String> phoneList = findPhone(predicate.get(i + 1));
+                for (int j = 0; j < phoneList.size(); j++) {
+                    if (!predicateList.contains(phoneList.get(j))) {
+                        predicateList.add(phoneList.get(j));
+                    }
+                }
+            }
+
+            if (predicates.equals(PREFIX_ADDRESS.getPrefix())) {
+                ArrayList<String> addressList = findAddress(predicate.get(i + 1));
+                for (int j = 0; j < addressList.size(); j++) {
+                    if (!predicateList.contains(addressList.get(j))) {
+                        predicateList.add(addressList.get(j));
+                    }
+                }
+            }
+
+            if (predicates.equals(PREFIX_EMAIL.getPrefix())) {
+                ArrayList<String> emailList = findEmail(predicate.get(i + 1));
+                for (int j = 0; j < emailList.size(); j++) {
+                    if (!predicateList.contains(emailList.get(j))) {
+                        predicateList.add(emailList.get(j));
+                    }
+                }
+            }
+
+            if (predicates.equals(PREFIX_TAG.getPrefix())) {
+                ArrayList<String> tagList = findTags(predicate.get(i + 1));
+                for (int j = 0; j < tagList.size(); j++) {
+                    if (!predicateList.contains(tagList.get(j))) {
+                        predicateList.add(tagList.get(j));
+                    }
+                }
+            }
+
+            if (predicates.equals(PREFIX_BIRTHDAY.getPrefix())) {
+                ArrayList<String> birthdayList = findBirthday(predicate.get(i + 1));
+                for (int j = 0; j < birthdayList.size(); j++) {
+                    if (!predicateList.contains(birthdayList.get(j))) {
+                        predicateList.add(birthdayList.get(j));
+                    }
+                }
+            }
+        }
+
+        NameContainsKeywordsPredicate predicates = new NameContainsKeywordsPredicate(predicateList);
+        model.updateFilteredPersonList(predicates);
+        return new CommandResult(getMessageForPersonListShownSummary(model.getFilteredPersonList().size()));
+    }
+
+```
+###### /java/seedu/address/logic/commands/FindCommand.java
+``` java
+    /**
+     * Search for persons that contain the {@String keyword} in their name
```

Comment:
Coding standard violation:exclamation: The first statement of the header comment does not follow the prescribed style. See point 4, 3rd bullet point of the [[Java Coding Standard: Comments](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#comments)]